### PR TITLE
Add support for retries

### PR DESCRIPTION
**Pull request by sethvargo**
/cc @pearkes @phinze

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,5 @@
 source "https://rubygems.org"
 
+gem "vault", github: "hashicorp/vault-ruby"
```
Comment (sethvargo) [inline Gemfile]: This is temporary - I'm gonna release a new vault-ruby

File: lib/vault/rails.rb [hunk 2]
```diff
@@ -192,12 +200,27 @@ def self.force_encoding(str)
...
+        attempts: self.retry_attempts,
+        base:     self.retry_base,
+        max_wait: self.retry_max_wait,
+      }
+
+      client.with_retries(exceptions, options) do |i, e|
+        if !e.nil?
+          log_warning "[vault-rails] (#{i}) An error occurred when trying to "
+            "communicate with Vault: #{e.message}"
         end
```
Comment (phinze) [inline lib/vault/rails.rb]: Should we just log anytime a retry is necessary?

**Comment by phinze** [comment]
LGTM - one wee question inline

File: lib/vault/rails.rb (hunk 2 above)
Comment (sethvargo) [inline lib/vault/rails.rb]: The block will still yield for the first request, but the exception will be nil. So the first request isn't actually a retry

Comment (phinze) [inline lib/vault/rails.rb]: Ah gotchya. :+1:

**Pull request merged**